docs/docs/working-with-video.md

```diff
@@ -131,9 +131,49 @@ export const pageQuery = graphql`
 
 It's super common to source video from YouTube, Twitch or Vimeo. But what if you want to host your own video and include it as HTML5 video?
 
-> This discussion is ongoing on GitHub, chime in with your ideas in the Gatsby issue [#3346 Create a special component for HTML5 videos](https://github.com/gatsbyjs/gatsby/issues/3346)
+To include your own video files that will work in multiple web browsers and platforms, you'll need to read up a bit on video extensions and codecs. We recommend MDN as a resource: [Media formats for HTML audio and video](https://developer.mozilla.org/en-US/docs/Web/HTML/Supported_media_formats). You may need video converter software to produce the necessary formats -- such as `.webm` and `.mp4` -- to support a range of devices and environments.
 
-To include your own video files that will work in multiple web browsers and platforms, you'll need to read up a bit on video extensions and codecs. We recommend MDN as a resource: [Media formats for HTML audio and video](https://developer.mozilla.org/en-US/docs/Web/HTML/Supported_media_formats). You may need video converter software to produce the necessary formats to support a range of devices and environments, such as `.webm` and `.mp4`.
+HTML5 provides the `<video>` media element for working with videos. Inside the `<video>` element, you can provide multiple `<source>` elements that serve as different file formats the video player can use, with each browser electing to use a format it supports.
+
+If you have a video called `dog.mp4` in your site under `src/assets/dog.mp4`, you can [include that video in your page with webpack](/docs/importing-assets-into-files) like you would other assets. Then reference it in a `<source>` element, which is wrapped by a `<video>` element:
+
+<!-- prettier-ignore -->
+```javascript:title=src/pages/index.js
+import React from "react"
+import DogVideo from "../assets/dog.mp4"
+
+export default () => (
+  <video controls style={{ width: "100%" }}>
+    <source src={DogVideo} type="video/mp4" /> // highlight-line
+  </video>
+)
+
+```
+
+The `controls` attribute on the `<video>` will provide buttons overlaid on the video to play/pause, adjust volume, and go full screen. Other attributes like `muted` can set audio to silent, or `poster` can display an image when the video isn't playing. Common attributes that you'd want to apply to multiple videos could be extracted into a custom React video component. A full list of `<video>` attributes can be found on [the MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#Attributes).
```

Comment:
I thought I worked on this already but maybe I stashed changes and didn't end up adding them - but we should take the opportunity to mention accessibility in video. This can go in a separate PR - but we should mention custom video player UI, transcripts, closed-captions, and audio description for video. This link is a great resource of the things we could mention: https://www.washington.edu/accessibility/videos/

There are also some Gatsby-specific things for audio in this post by Lindsey Kopacz: https://www.a11ywithlindsey.com/blog/a11y-lessons-learned-starting-podcast

I have a few ideas:
- Rename this page to include Working with Video and Audio, and add a few more audio details
- Mention accessibility on this video page and link to a more general page, like [Importing Media Content](https://www.gatsbyjs.org/docs/importing-media-content/)

I'd love to hear your thoughts on this too, @amberleyromo!

Reply:
Would it make sense to just have a separate page for Working with Audio? I guess some things for accessibility overlap but maybe a lot of that could go on a more general page like Importing Media Content.

I'll add some more info about video in this PR though so it doesn't just sit around waiting in an issue again.